.github/workflows/test.yml

```diff
@@ -143,17 +143,22 @@ jobs:
         id: get-commit-metadata
         shell: bash
         run: |
-          echo "timestamp=$(git show --format='%ct' --no-patch)" >> "${GITHUB_OUTPUT}"
           if [ ${{ github.event_name }} = 'push' ]; then
             echo "branch=${{ github.ref_name }}" >> "${GITHUB_OUTPUT}"
+            # Ignore top most commit which just adds CI files.
+            commit="$(git rev-parse HEAD^)"
```

Comment:
I wonder if we could get kpd to add a tag to the commits it generates, this way we could strip the topmost commits with the tag present.
Reasoning is that the day we add a new commit via KPD, this will break.

Reply:
The problem I see is that then we rely on the tag being present itself and the breakage of it not being there would be more severe (`HEAD^` is always present, but it may reference a non-upstream commit or one earlier than expected; neither conditions really break anything here, as I tried to explain in the summary). We can guard against the tag not being there, but then we are back at square one on the fallback path.

Ideally we'd need neither and have a way to identify something as upstream. We could do tricks such as take the first commit with a `Link:` tag in its summary, but I felt that's not more stable either. My preference would be to use something like `git-merge-base` against a known branch, but I haven't been able to make that work -- the clone/checkout we work on didn't have a whole lot of stuff to work with. I can play around some more and see if I can improve it.

Reply:
Going to land the change first, though, as I still think it's an improvement.